python/semantic_kernel/connectors/ai/function_calling_utils.py

```diff
@@ -95,3 +96,30 @@ def merge_function_results(
             items=items,
         )
     ]
+
+
+def merge_streaming_function_results(
+    messages: list[ChatMessageContent | StreamingChatMessageContent],
+    ai_model_id: str,
+) -> list[StreamingChatMessageContent]:
+    """Combine multiple streaming function result content types to one streaming chat message content type.
+
+    This method combines the FunctionResultContent items from separate StreamingChatMessageContent messages,
+    and is used in the event that the `context.terminate = True` condition is met.
+
+    Args:
+        messages: The list of streaming chat message content types.
+        ai_model_id: The AI model ID.
+
+    Returns:
+        The combined streaming chat message content type.
+    """
+    items: list[Any] = []
+    for message in messages:
+        items.extend([item for item in message.items if isinstance(item, FunctionResultContent)])
+
+    # If we want to be able to support adding the streaming message chunks together, then the author role needs to be
+    # `Assistant```, as the `Tool` role will cause the add method to break.
+    return [
+        StreamingChatMessageContent(role=AuthorRole.ASSISTANT, items=items, choice_index=0, ai_model_id=ai_model_id)
```

Comment:
This is going to cause issue in some connectors if the function result is not with a role that they expect. For example, if someone uses a connector at the beginning then later switch to another to continue the same chat history.

Why couldn't streaming messages with the tool role be added together?

Reply:
It seems there was a conscious decision to not allow to streaming chat message content chunks to be added together if their roles are different: https://github.com/microsoft/semantic-kernel/blob/62a50f32cf140b24876517219726a28465ef640e/python/semantic_kernel/contents/streaming_chat_message_content.py#L171

Reply:
That's ok, no? These are tool messages; we don't want them to be concatenated with the assistant message.

Reply:
Synced offline - will keep the Streaming CMC for FRC as a TOOL role.